webvtt.go

```diff
@@ -306,45 +305,64 @@ func parseTextWebVTT(i string) (o Line) {
 	// Create tokenizer
 	tr := html.NewTokenizer(strings.NewReader(i))
 
+	webVTTTagStack := make([]*WebVTTTag, 64)
+	webVTTTagStackIndex := -1
+
 	// Loop
-	italic := false
+	loopCount := 0
 	for {
 		// Get next tag
 		t := tr.Next()
-
+		loopCount++
 		// Process error
 		if err := tr.Err(); err != nil {
 			break
 		}
 
 		switch t {
 		case html.EndTagToken:
-			// Parse italic
-			if bytes.Equal(tr.Raw(), bytesWebVTTItalicEndTag) {
-				italic = false
-				continue
+			// Pop the top of stack if we meet end tag
+			if webVTTTagStackIndex >= 0 {
+				webVTTTagStackIndex--
 			}
 		case html.StartTagToken:
-			// Parse voice name
-			if matches := webVTTRegexpStartTag.FindStringSubmatch(string(tr.Raw())); len(matches) > 3 {
-				if s := strings.TrimSpace(matches[3]); s != "" {
-					o.VoiceName = s
+			if matches := webVTTRegexpTag.FindStringSubmatch(string(tr.Raw())); len(matches) > 4 {
+				tagName := matches[2]
+
+				var classes []string
+				if matches[3] != "" {
+					classes = strings.Split(strings.Trim(matches[3], "."), ".")
 				}
-				continue
-			}
 
-			// Parse italic
-			if bytes.Equal(tr.Raw(), bytesWebVTTItalicStartTag) {
-				italic = true
-				continue
+				annotation := ""
+				if matches[4] != "" {
+					annotation = strings.TrimSpace(matches[4])
+				}
+
+				// Only support voicename if <v> appears in the begining of the line
```

Comment:
Why only support voicename if it's present at the beginning and not elsewhere? 🤔

Reply:
The .VoiceName was added to support converting from webvtt to ssa/ass format, and ssa/ass only support a voice name for entire line, as far as I know. However, WebVTT support multiple voice in the same line, like this:
```
Someone's saying <v Joe> So am I </v> <v John> Me too </v> <i>Joe and John talk at the same time</i>
```
And With the current behavior "John" will be the voice for ssa/ass, with this behavior no voice will be picked because both are in correct for the voice name of the line.


For preserve `<v>` tag in WebVTT format, the story is more complex, consider this example:
```
<v Ana> I think this dish is <u>fabulous</u><v> 
```
When applied the PR's new logic, the outcome is:
```
<v Ana> I think this dish is</v> <v Ana><u>fabulous</u></v>
```
I don't know if this styling is the same with the original one and looked for a way to verify, but hopeless. So I decided to postpone until I can make sure that we will now get into any trouble because of the outcome like that.

Reply:
Thanks for the clarification 👍 

Instead of creating `loopCount`, could you replace the `if` condition with `if o.VoiceName == "" && tagName == "v"`?

Reply:
For the example that I gave above:
```
Someone's saying <v Joe> So am I </v> <v John> Me too </v> <i>Joe and John talk at the same time</i>
```
With your condition, `Joe` will be chosen, but `John` also talk something. So I think choosing `Joe` or `John` is also wrong, and eliminate the voice is a better idea? Or we need to keep one of them?  I refer choosing no one as voice name, since no matter who we choose, we will be incorrect.

Reply:
Thing is, if we choose no voice, we're throwing data away and we're kind of also wrong 🤔 I'd be OK storing the first voice we find and add a TODO to process other voices.

Otherwise I guess this condition would do the trick:

```go
if  tagName == "v" {
  if o.VoiceName == "" {
    o.VoiceName = annotation
  } else {
    // Clean voice name so that we store no voice
    o.VoiceName = ""
  }
}
```

Reply:
well, if we have three voice, the third voice will be chosen then :thinking: . Alright, I will choose the first voice for the voice name of the whole line, it would be good if we print some thing out to warn user that we currently ignore other voices in the the same line.

Reply:
👍